optimizely/notification/center.go

```diff
@@ -0,0 +1,59 @@
+/****************************************************************************
+ * Copyright 2019, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+package notification
+
+import "fmt"
+
+// Center handles all notification listeners. It keeps track of the Manager for each type of notification.
+type Center interface {
+	AddHandler(Type, func(interface{})) (int, error)
+	Send(Type, interface{}) error
+}
+
+// DefaultCenter contains all the notification managers
+type DefaultCenter struct {
+	managerMap map[Type]Manager
+}
+
+// NewNotificationCenter returns a new notification center
+func NewNotificationCenter() *DefaultCenter {
+	decisionNotificationManager := &AtomicManager{}
+	managerMap := make(map[Type]Manager)
+	managerMap[Decision] = decisionNotificationManager
+	return &DefaultCenter{
+		managerMap: managerMap,
+	}
+}
+
+// AddHandler adds a handler for the given notification type
+func (c *DefaultCenter) AddHandler(notificationType Type, handler func(interface{})) (int, error) {
+	if manager, ok := c.managerMap[notificationType]; ok {
+		return manager.AddHandler(handler)
+	}
+
+	return -1, fmt.Errorf("No notification manager found for type %s", notificationType)
+}
+
+// Send sends the given notification payload to all listeners of type
+func (c *DefaultCenter) Send(notificationType Type, notification interface{}) error {
+	if manager, ok := c.managerMap[notificationType]; ok {
+		manager.Send(notification)
+		return nil
+	}
+
+	return fmt.Errorf("No notification manager found for type %s", notificationType)
```

Comment:
Lets be consistent in terminology and say notification handler and not manager.

Reply:
Same as above